From 196d6da6ce31ade2fdbe783f243fc9997adab055 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Mon, 17 Jun 2013 13:43:33 +0200
Subject: [PATCH 183/628] xilinx: Use platform_{get,set}_drvdata()

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit 3bd0d0850bf5f96cda184200997264d31a51dfd8

Based on mainline commit:
"serial: use platform_{get,set}_drvdata()"
(sha1: 696faedd616e202f5c510cd03dcc8853c11ca6db)

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
[cherry picked the devcfg driver modification only]
Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/char/xilinx_devcfg.c |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/char/xilinx_devcfg.c b/drivers/char/xilinx_devcfg.c
index c9edc55..96818a6 100644
--- a/drivers/char/xilinx_devcfg.c
+++ b/drivers/char/xilinx_devcfg.c
@@ -1864,7 +1864,7 @@ static int xdevcfg_drv_probe(struct platform_device *pdev)
 		goto failed0;
 	}
 
-	dev_set_drvdata(&pdev->dev, drvdata);
+	platform_set_drvdata(pdev, drvdata);
 
 	if (!request_mem_region(regs_res->start,
 				regs_res->end - regs_res->start + 1,
@@ -2030,7 +2030,7 @@ static int xdevcfg_drv_remove(struct platform_device *pdev)
 	struct xdevcfg_drvdata *drvdata;
 	struct resource *res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
 
-	drvdata = dev_get_drvdata(&pdev->dev);
+	drvdata = platform_get_drvdata(pdev);
 
 	if (!drvdata)
 		return -ENODEV;
@@ -2050,7 +2050,6 @@ static int xdevcfg_drv_remove(struct platform_device *pdev)
 	clk_unprepare(drvdata->clk);
 	clk_put(drvdata->clk);
 	kfree(drvdata);
-	dev_set_drvdata(&pdev->dev, NULL);
 
 	return 0;		/* Success */
 }
-- 
1.7.5.4

